app/utils/build/__init__.py

```diff
@@ -524,12 +524,8 @@ def import_single_build(json_obj, db_options, base_path=utils.BASE_PATH):
     git_branch = utils.clean_branch_name(git_branch)
 
     if (utils.valid_name(job) and utils.valid_name(kernel)):
-        # New directory structure:
-        # $job/$branch/$kernel/$arch/$defconfig/$environment
-
-        parent_dir = os.path.join(base_path, job, git_branch, kernel, arch)
-        build_dir = os.path.join(
-            parent_dir, defconfig_full or defconfig, build_environment)
+        file_server_resource = j_get(models.FILE_SERVER_RESOURCE_KEY)
```

Comment:
So this is now a mandatory key so should be checked as such.
If it's not mandatory, it should fall back on the previous behavior.

Reply:
Good point, I've changed that and updated the branch.  It should be mandatory as there's no reason for not using afaict.